File: C/beginner/1156.c
// https://judge.beecrowd.com/pt/problems/view/1156
// Sequência S II

// Libraries
#include <stdio.h>

// Main
int main(){

    // Declaring Variables
    double num, den, sum;
    
    // Starting variables
    sum = 1.0;
    num = 1.0;
    den = 1.0;
    
    // Iterating
    for(int i = 1; i <= 39; i += 2){
        num += 2;
        den *= 2;
        sum += (num / den);

    }

    // Showing
    printf("%.2lf\n", sum);

    return 0;
}

// By Fernando Serra
// https://github.com/fernandosserra

File: Haskell/beginner/1156.hs
-- https://judge.beecrowd.com/pt/problems/view/1156
-- Sequência S II

-- Importing Printf
import Text.Printf (printf)

-- progressiveFractions
progressiveFractions :: Double
progressiveFractions = sum $ zipWith (/) [fromIntegral i | i <- [1,3 .. 39]] [2.0^j | j <- [0..19]]

-- main
main :: IO ()
main = printf "%.2f\n" progressiveFractions

-- By Fernando Serra
-- https://github.com/fernandosserra